cmd/vic-machine/validator.go

```diff
@@ -0,0 +1,210 @@
+	"github.com/vmware/vic/metadata"
+	"github.com/vmware/vic/pkg/errors"
+	"github.com/vmware/vic/pkg/vsphere/session"
+
+	"golang.org/x/net/context"
+)
+
+type Validator struct {
+	TargetPath            string
+	DatacenterName        string
+	ClusterPath           string
+	ResourcePoolPath      string
+	ImageStorePath        string
+	ExternalNetworkPath   string
+	BridgeNetworkPath     string
+	BridgeNetworkName     string
+	ManagementNetworkPath string
+	ManagementNetworkName string
+
+	Session *session.Session
+	Context context.Context
+}
+
+func NewValidator() *Validator {
+	return &Validator{}
+}
+
+func (v *Validator) Validate(input *Data) (*metadata.VirtualContainerHostConfigSpec, error) {
+
+	vchConfig := &metadata.VirtualContainerHostConfigSpec{}
+	vchConfig.VCHSize.CPU.Limit = input.numCPUs
+	vchConfig.VCHSize.Memory.Limit = input.memoryMB
+	vchConfig.Name = input.displayName
+
+	resources := strings.Split(input.computeResourcePath, "/")
+	if len(resources) < 2 || resources[1] == "" {
+		err := errors.Errorf("Could not determine datacenter from specified -compute path, %s", input.computeResourcePath)
+		return nil, err
+	}
+	v.DatacenterName = resources[1]
+	v.ClusterPath = strings.Split(input.computeResourcePath, "/Resources")[0]
+
+	if v.ClusterPath == "" {
+		err := errors.Errorf("Could not determine cluster from specified -compute path, %s", input.computeResourcePath)
+		return nil, err
+	}
+
+	v.ResourcePoolPath = input.computeResourcePath
+	v.ImageStorePath = fmt.Sprintf("/%s/datastore/%s", v.DatacenterName, input.imageDatastoreName)
+
+	v.TargetPath = fmt.Sprintf("%s:%s@%s/sdk", input.user, *input.passwd, input.target)
+	vchConfig.Target = v.TargetPath
+	vchConfig.Insecure = input.insecure
+
+	v.ExternalNetworkPath = fmt.Sprintf("/%s/network/%s", v.DatacenterName, input.externalNetworkName)
+	v.BridgeNetworkPath = fmt.Sprintf("/%s/network/%s", v.DatacenterName, input.bridgeNetworkName)
+	v.BridgeNetworkName = input.bridgeNetworkName
+	if input.managementNetworkName != "" {
+		v.ManagementNetworkPath = fmt.Sprintf("/%s/network/%s", v.DatacenterName, input.managementNetworkName)
+		v.ManagementNetworkName = input.managementNetworkName
+	}
+	vchConfig.ImageStoreName = input.imageDatastoreName
+	vchConfig.DatacenterName = v.DatacenterName
+	vchConfig.ClusterPath = v.ClusterPath
+
+	if err := v.validateConfiguration(input, vchConfig); err != nil {
+		return nil, err
+	}
+	return vchConfig, nil
+}
+
+func (v *Validator) validateConfiguration(input *Data, vchConfig *metadata.VirtualContainerHostConfigSpec) error {
+	log.Infof("Validating supplied configuration")
+
+	var err error
+	v.Context = context.TODO()
+	sessionconfig := &session.Config{
+		Service:        v.TargetPath,
+		Insecure:       input.insecure,
+		DatacenterPath: v.DatacenterName,
+		ClusterPath:    v.ClusterPath,
+		DatastorePath:  v.ImageStorePath,
+		PoolPath:       v.ResourcePoolPath,
+	}
+
+	v.Session, err = session.NewSession(sessionconfig).Create(v.Context)
```

Comment:
After discussion with George, will keep as it is right now, cause the objects in Session is leveraged in both validator and dispatcher.
We'll revisit it after govmomi gateway is ready.